RecoJets/JetPlusTracks/plugins/JetPlusTrackProducer.cc

```diff
@@ -91,25 +102,122 @@ void JetPlusTrackProducer::produce(edm::Event& iEvent, const edm::EventSetup& iS
   edm::Handle<edm::View<reco::CaloJet> > jets_h;
   iEvent.getByToken(input_jets_token_, jets_h);
 
-  //  auto pOut = std::make_unique<reco::CaloJetCollection>();
+  //JetPlusTrackAddonSeed
+  edm::Handle<edm::View<reco::CaloJet> > addjets_h;
+  iEvent.getByToken(input_addjets_token_, addjets_h);
+
+  //std::cout<<" Additional Calojets "<<addjets_h->size()<<std::endl;
+
+  edm::Handle<edm::View<reco::TrackJet> > jetsTrackJets;
+  iEvent.getByToken(input_trackjets_token_, jetsTrackJets);
+
+  //std::cout<<" Additional Trackjets "<<jetsTrackJets->size()<<std::endl;
+
+  edm::Handle<std::vector<reco::TrackExtrapolation> > iExtrapolations;
+  iEvent.getByToken(mExtrapolations, iExtrapolations);
+
+  edm::RefProd<reco::CaloJetCollection> pOut1RefProd = iEvent.getRefBeforePut<reco::CaloJetCollection>();
+  edm::Ref<reco::CaloJetCollection>::key_type idxCaloJet = 0;
+
   auto pOut = std::make_unique<reco::JPTJetCollection>();
+  auto pOut1 = std::make_unique<reco::CaloJetCollection>();
+
+  double scaleJPT = 1.;
+  std::vector<reco::JPTJet> theJPTJets;
+  if (jetsTrackJets.isValid()) {
+    if (!jetsTrackJets->empty()) {
+      for (unsigned ijet = 0; ijet < jetsTrackJets->size(); ++ijet) {
+        const reco::TrackJet* jet = &(*(jetsTrackJets->refAt(ijet)));
+        int icalo = -1;
+        if (addjets_h.isValid()) {
+          for (unsigned i = 0; i < addjets_h->size(); ++i) {
+            const reco::CaloJet* oldjet = &(*(addjets_h->refAt(i)));
+            double dr = reco::deltaR<double>(jet->eta(), jet->phi(), oldjet->eta(), oldjet->phi());
+            if (dr <= dRcone) {
+              icalo = i;
+            }
+          }  // Calojets
+        }
+        if (icalo < 0)
+          continue;
+        const reco::CaloJet* mycalo = &(*(addjets_h->refAt(icalo)));
+        //     std::cout<<" Additional CaloJet "<<mycalo->pt()<<" "<<mycalo->eta()<<
+        //                                   " "<<mycalo->phi()<<std::endl;
+        std::vector<edm::Ptr<reco::Track> > tracksinjet = jet->tracks();
+        reco::TrackRefVector tracksincalo;
+        reco::TrackRefVector tracksinvert;
+        for (std::vector<edm::Ptr<reco::Track> >::iterator itrack = tracksinjet.begin(); itrack != tracksinjet.end();
+             itrack++) {
+          for (std::vector<reco::TrackExtrapolation>::const_iterator xtrpBegin = iExtrapolations->begin(),
+                                                                     xtrpEnd = iExtrapolations->end(),
+                                                                     ixtrp = xtrpBegin;
+               ixtrp != xtrpEnd;
+               ++ixtrp) {
+            if (ixtrp->positions().empty())
+              continue;
+            double mydphi = deltaPhi(ixtrp->track()->phi(), (**itrack).phi());
+            if (fabs(ixtrp->track()->pt() - (**itrack).pt()) > 0.001 ||
+                fabs(ixtrp->track()->eta() - (**itrack).eta()) > 0.001 || mydphi > 0.001)
+              continue;
```

Comment:
this deep-nested cross-referencing via eta,phi will likely have to be redone (pending timing cost review)